### PR TITLE
trivial: Document the expected plugin firmware formats

### PR DESCRIPTION
**Pull request by hughsie**

### CODE REVIEW

File: plugins/dell-dock/README.md [hunk 1]
```diff
@@ -25,6 +25,12 @@ When this plugin is used, devices present in other plugins may be shown in
 the topology of this dock.  This is intentional as this plugin works together
 with those plugins to manage the flashing of all components.
 
+Firmware Format
+---------------
+
+The daemon will decompress the cabinet archive and extract a firmare blob in
+an unspecified binary file format.
```
Comment (superm1) [inline plugins/dell-dock/README.md]: This is actually a bunch of blobs and XML files

File: plugins/colorhug/README.md [hunk 2]
```diff
@@ -11,6 +11,12 @@ accurate color matching.
 ColorHug versions 1 and 2 support a custom HID-based flashing protocol, but
 version 3 (ColorHug+) has now switched to DFU.
 
+Firmware Format
+---------------
+
+The daemon will decompress the cabinet archive and extract a firmare blob in
+an packed binary file format.
```
Comment (superm1) [inline plugins/colorhug/README.md]: /an/a/

File: plugins/uefi/README.md [hunk 3]
```diff
@@ -11,6 +11,12 @@ With the UpdateCapsule boot service it can be used to update system firmware.
 If you don't want or need this functionality you can use the
 `-Dplugin_uefi=false` option.
 
+Firmware Format
+---------------
+
+The daemon will decompress the cabinet archive and extract a firmare blob in
+EFI executable (capsule) file format.
```
Comment (superm1) [inline plugins/uefi/README.md]: Technically this is not executable

Comment (hughsie) [reply plugins/uefi/README.md]: `blob in an EFI Capsule file format`? Is Capsule a file format?

File: plugins/dell-dock/README.md (hunk 1 above)
Comment (hughsie) [reply plugins/dell-dock/README.md]: so something like `The daemon will decompress the cabinet archive and extract several firmware blobs with an unspecified binary file format.`?

File: plugins/uefi/README.md (hunk 3 above)
Comment (superm1) [reply plugins/uefi/README.md]: Well the structure of a capsule is defined by the EFI specs.
Section 22.2.2: " DEFINED FIRMWARE MANAGEMENT PROTOCOL DATA
CAPSULE STRUCTURE"
https://www.uefi.org/sites/default/files/resources/UEFI%20Spec%202_6.pdf

**Pull request merged**